src/Reflection/ParameterReflection.php

```diff
@@ -19,16 +21,18 @@ class ParameterReflection extends ReflectionParameter implements ReflectionInter
     /**
      * Get declaring class reflection object
      *
-     * @return ClassReflection
+     * @return ClassReflection|null
      */
     #[ReturnTypeWillChange]
     public function getDeclaringClass()
     {
-        $phpReflection     = parent::getDeclaringClass();
-        $laminasReflection = new ClassReflection($phpReflection->getName());
-        unset($phpReflection);
+        $reflection = parent::getDeclaringClass();
 
-        return $laminasReflection;
+        if (! $reflection) {
```

Comment:
A more specific check here would be preferable from my POV, but again I leave this to your decision.

Reply:
Similar to above checks, `false|object` here OK with me for now :+1: